test/jdk/java/awt/image/multiresolution/MultiDisplayTest.java

```diff
@@ -21,63 +21,106 @@
-  @library /test/lib
-  @build jdk.test.lib.Platform
-  @run applet/manual=yesno MultiDisplayTest.html
-*/
-
-
-import java.applet.Applet;
-import java.awt.*;
-import java.awt.event.*;
-import java.awt.image.*;
+ * @test
+ * @bug 8142861 8143062 8147016
+ * @library /java/awt/regtesthelpers /test/lib
+ * @build PassFailJFrame jdk.test.lib.Platform
+ * @requires (os.family == "windows" | os.family == "mac")
+ * @summary Check if multiresolution image behaves properly
+ *         on HiDPI + non-HiDPI display pair.
+ * @run main/manual MultiDisplayTest
+ */
 
-import jdk.test.lib.Platform;
+public class MultiDisplayTest {
+    private static JFrame frame;
+    private static final String INSTRUCTIONS =
+            """
+             The test requires two-display configuration, where
+
+             - 1st display is operating in HiDPI mode;
+             - 2nd display is non-HiDPI.
+
+             In other cases please simply push "Pass".
+
+             To run test please push "Start".
+
+             Then drag parent / child to different displays and check
+             that the proper image is shown for every window
+             (must be "black 1x" for non-HiDPI and "blue 2x" for HiDPI).
 
-public class MultiDisplayTest extends Applet {
+             Please try to drag both parent and child,
+             do it fast several times and check if no artefacts occur.
+
+             Try to switch display resolution (high to low and back).
+
+             For Mac OS X please check also the behavior for
+             translucent windows appearing on the 2nd (non-active) display
+             and Mission Control behavior.
+
+             Close the windows.
+
+             In case if no issues occur please push "Pass", otherwise "Fail"
+            """;
 
     private static final int W = 200, H = 200;
 
     private static final BaseMultiResolutionImage IMG =
         new BaseMultiResolutionImage(new BufferedImage[]{
         generateImage(1, Color.BLACK), generateImage(2, Color.BLUE)});
 
-    private static boolean checkOS() {
-        return Platform.isWindows() || Platform.isOSX();
+    public static void main(String[] args) throws Exception {
+        if (!checkOS()) {
+            throw new SkippedException("Invalid OS." +
+                    "Please run test on either Windows or MacOS");
+        }
+        PassFailJFrame
+                .builder()
+                .title("MultiDisplayTest Instructions")
+                .instructions(INSTRUCTIONS)
+                .rows((int) INSTRUCTIONS.lines().count() + 2)
+                .columns(40)
+                .testUI(MultiDisplayTest::createAndShowGUI)
+                .build()
+                .awaitAndCheck();
     }
 
-    public void init() { this.setLayout(new BorderLayout()); }
-
-    public void start() {
-
+    public static JFrame createAndShowGUI() {
+        frame = new JFrame("MultiDisplayTest");
+        frame.setLayout(new BorderLayout());
         Button b = new Button("Start");
```

Comment:
Just noticed that the `frame` is not used in test itself, so I guess we can get rid of it, and make this method return the "Start" `JButton`, and use it with `splitUIBottom`/`splitUIRight` instead of `testUI`

Reply:
Good point. I Removed the `frame`. Changed `button` to `JButton`. Did everything else accordingly.